### ISSUE TITLE
Packer not attempting any SSH connections on VirtualBox if nic1 is hostonlyadapter

### ISSUE THREAD

**Issue by anuwan94** [hashicorp/packer#9048]
**Description**
packer is not attempting SSH connections if the nic1 interface of the VM is hostonlyadapter, it will work if I "trick" it with natnetwork.
`
            ["modifyvm", "{{.Name}}", "--natnet1", "192.168.0/24"],
`
But not with hostonlyadapter.

**Environment**
Up-to-date Arch linux with: 
```
pacman -Q linux packer virtualbox
linux 5.6.3.arch1-1
packer 1.5.5-1
virtualbox 6.1.4-6
```
Also the hostonly interface is up and working correctly.
```
ip addr sh vboxnet0
4: vboxnet0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc fq_codel state DOWN group default qlen 1000
    link/ether 0a:00:27:00:00:00 brd ff:ff:ff:ff:ff:ff
    inet 192.168.0.1/24 brd 192.168.0.255 scope global vboxnet0
       valid_lft forever preferred_lft forever
    inet6 fe80::800:27ff:fe00:0/64 scope link 
       valid_lft forever preferred_lft forever
```

**Tests**
I'm running the same template and **it's working** : 
```
==> virtualbox-iso: Typing the boot command...
==> virtualbox-iso: Using ssh communicator to connect: 127.0.0.1
==> virtualbox-iso: Waiting for SSH to become available...
==> virtualbox-iso: Connected to SSH!
==> virtualbox-iso: Uploading VirtualBox version info (6.1.4)
==> virtualbox-iso: Provisioning with shell script: provisioning/
```
until finally the image is generated, however if I just change the nic type and specify some more SSH options (to help out packer to connect) it will not work : 
Below what changes from the previous (working) template : 

```
        "ssh_host": "192.168.0.2",
        "ssh_port": "22",
        "vboxmanage": [
            ["modifyvm", "{{.Name}}", "--nic1", "hostonly"],
            ["modifyvm", "{{.Name}}", "--hostonlyadapter1", "vboxnet0"]
```
It's **not working** anymore : 
```
==> virtualbox-iso: Typing the boot command...
==> virtualbox-iso: Using ssh communicator to connect: 192.168.0.2
==> virtualbox-iso: Waiting for SSH to become available...
```

**Important**
I kept monitoring both vboxnet0 interface as well as with **any** interface for SSH traffic : 
```
sudo tcpdump -nnvvi any port 22
sudo tcpdump -nnvvi vboxnet0 port 22
```
And no SYN request was recorded on any interface for destination port 22.
I checked as well for ARP traffic, and was given when the virtual machine was up.
Needless to say, that if I SSH'ed directly from the Host with : 
ssh root@192.168.0.2 
**It worked** and the virtual machine was reachable.

**Comment by sylviamoss** [comment]
Hey there! Thanks for reaching out. 
I think this is happening because the ssh port forwarding, which happens before vboxmanage commands, forces the NAT network interface [[code here]](https://github.com/hashicorp/packer/blob/master/builder/virtualbox/common/step_port_forwarding.go#L72), so I guess when you change the interface you invalidate the port forwarding. 

Can you tell me more about why are you setting a different network interface? With that our team can discuss if we should maybe add a new config for this or something else.

**Comment by anuwan94** [comment]
Thanks for the feedback, I was able to solve this with 
`"skip_nat_mapping": true`
As for your question,  the users who work with firewalls / utm systems often have at least 2 networks (lan / wan) where the boot starts up the system with a preset IP on the LAN on it's own network.
Our current options now would either via NAT where the NAT network  should have the same subnet as the LAN of the firewall, or the one I tried above where I created a hostonly adapter for the LAN in order to configure.

This works, but often that LAN should be transferred on a internal-lan network , so having packer able to SSH to a new IP during provisioning would be very nice since for example.

You finished the first stage and packer SSH'ed on the firewall and started provisioning. With one of the provisioning files the first interface is set to LAN and from virtaulbox as an internal network.
Under normal means packer would loose the ability to connect back via that interface since now it's on an internal network which as we know is not reachable from the host directly.
So , specifying on this phase new details for a new connection would be really great but maybe it's not possible.
